bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/NashornScriptEngineFactory.java

```diff
@@ -72,4 +80,25 @@ public void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValu
             logger.error("ScriptException while importing scope: {}", ex.getMessage());
         }
     }
+
+    @Override
+    public @Nullable ScriptEngine createScriptEngine(String scriptType) {
+        try {
+            for (javax.script.ScriptEngineFactory f : ENGINE_MANAGER.getEngineFactories()) {
+                List<String> mimeTypes = f.getMimeTypes();
+                List<String> extensions = f.getExtensions();
+
+                if (mimeTypes.contains(scriptType) || extensions.contains(scriptType)) {
+                    Method method = f.getClass().getMethod("getScriptEngine", ClassLoader.class);
+                    return (ScriptEngine) method.invoke(f, NashornScriptEngineFactory.class.getClassLoader());
```

Comment:
If the classloader that creates the component is also usable, you might be able to get rid of this reflection and reuse the same `ScriptFactory` in `createScriptEngine` like I did with Groovy by just assigning the `ScriptFactory` to a field.

See: [GroovyScriptEngineFactory](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.automation.groovyscripting/src/main/java/org/openhab/automation/groovyscripting/internal/GroovyScriptEngineFactory.java)

Reply:
That's what I did initially, but I got resolution failures (https://ci.openhab.org/job/PR-openHAB-Core/2682/console):
`missing requirement [org.openhab.core.automation.module.script/3.0.0.202011051754] osgi.wiring.package; filter:="(osgi.wiring.package=jdk.nashorn.api.scripting)"]]`
and I didn't know how to solve them 😞

Reply:
It's probably because it is considered to be internal JDK package and it may not be available in each JVM implementation. Since it is also deprecated it's probably best not to depend on it also because Nashorn has been removed since Java 15.